### PR TITLE
Force Netty to use an internal logger based on JBoss logging

### PR DESCRIPTION
**Pull request by galderz**
With this, exceptions appear ok:

```
[DEBUG] [io.quarkus.builder] Dependency of "io.quarkus.elytron.security.deployment.SecurityDeploymentProcessor#registerAdditionalBeans" finished; 0 remaining
[DEBUG] [io.quarkus.builder] Starting step "io.quarkus.elytron.security.deployment.SecurityDeploymentProcessor#registerAdditionalBeans"
[DEBUG] [io.netty.util.internal.PlatformDependent0] direct buffer constructor: unavailable
java.lang.UnsupportedOperationException: Reflective setAccessible(true) disabled
    at io.netty.util.internal.ReflectionUtil.trySetAccessible (ReflectionUtil.java:31)
    at io.netty.util.internal.PlatformDependent0$4.run (PlatformDependent0.java:224)
    at java.security.AccessController.doPrivileged (Native Method)
    at io.netty.util.internal.PlatformDependent0.<clinit> (PlatformDependent0.java:218)
    at io.netty.util.internal.PlatformDependent.isAndroid (PlatformDependent.java:213)
    at io.netty.util.internal.PlatformDependent.<clinit> (PlatformDependent.java:81)
    at io.netty.buffer.UnpooledByteBufAllocator.<clinit> (UnpooledByteBufAllocator.java:37)
    at io.netty.buffer.Unpooled.<clinit> (Unpooled.java:73)
    at io.netty.handler.codec.http.HttpObjectEncoder.<clinit> (HttpObjectEncoder.java:53)
    at java.lang.Class.forName0 (Native Method)
    at java.lang.Class.forName (Class.java:315)
    at io.quarkus.netty.deployment.NettyProcessor.build (NettyProcessor.java:50)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at io.quarkus.deployment.ExtensionLoader$1.execute (ExtensionLoader.java:768)
    at io.quarkus.builder.BuildContext.run (BuildContext.java:415)
    at org.jboss.threads.ContextClassLoaderSavingRunnable.run (ContextClassLoaderSavingRunnable.java:35)
    at org.jboss.threads.EnhancedQueueExecutor.safeRun (EnhancedQueueExecutor.java:2011)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask (EnhancedQueueExecutor.java:1535)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1426)
    at java.lang.Thread.run (Thread.java:834)
    at org.jboss.threads.JBossThread.run (JBossThread.java:479)
[DEBUG] [io.quarkus.builder] Dependency of "io.quarkus.arc.deployment.BeanArchiveProcessor#build" finished; 8 remaining
```

As mentioned on #3928, there might be easier ways to get this fixed. This is one way.

### CODE REVIEW

**Comment by emmanuelbernard** [comment]
Since there is rework at these layers, asking @stuartwdouglas for a quick review.

**Comment by gsmet** [comment]
@galderz Is it me or the exceptions are not consistent with our usual JBoss Logging formatter? Or did you tweak it?

**Comment by galderz** [comment]
> @galderz Is it me or the exceptions are not consistent with our usual JBoss Logging formatter? Or did you tweak it?

What do you mean exactly? The exception is as logged to the output. If you remove this patch and set augmentor thread pool to 1, you'll see the same.

**Comment by dmlloyd** [comment]
The Mojo logger passes things through to Maven, so it's going to be using that format, not our default JBoss LogManager format.

**Review by dmlloyd — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
There is no guarantee when that build step will be run, so its possible that these changes may only take effect half way through the build. I think it would be better to install it in a static initialiser block, as AFAIK the BuildStep classes are all initialised early enough to make this guaranteed to work.

Also this is done at augment time, do we need something similar for runtime?

**Comment by galderz** [comment]
> There is no guarantee when that build step will be run, so its possible that these changes may only take effect half way through the build. I think it would be better to install it in a static initialiser block, as AFAIK the BuildStep classes are all initialised early enough to make this guaranteed to work.

Makes sense, I'll try to address that.

> Also this is done at augment time, do we need something similar for runtime?

I'm not sure if it's such an issue at runtime, because I guess that can run outside of Maven? Image augmentation for native image happens for sure at Maven build time, which is what I was trying to debug.

**Review by gsmet — DISMISSED** [review]
This one shouldn't be merged for now. See Stuart's comment.

**Comment by galderz** [comment]
@stuartwdouglas @gsmet updated PR

File: extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java [hunk 1]
```diff
@@ -137,4 +138,10 @@ void createExecutors(BuildProducer<RuntimeBeanBuildItem> runtimeBeanBuildItemBui
                 .build());
     }
 
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void setJBossInternalLoggerFactory() {
```
Comment (stuartwdouglas) [inline extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java]: This was not what I meant (and should fail as it is not valid to have a @Record method that does not inject a recorder).

I just meant a static block to make sure it is run when this class is loaded:

  static {
    InternalLoggerFactory.setDefaultFactory(new JBossNettyLoggerFactory());
  }

Anything in a build step has the possibility that other build steps will be run first.

Comment (galderz) [reply extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java]: > This was not what I meant (and should fail as it is not valid to have a @record method that does not inject a recorder).

Ok... 🙈

There's [an example](https://github.com/quarkusio/quarkus/blob/master/core/test-extension/deployment/src/main/java/io/quarkus/extest/deployment/TestProcessor.java#L231) in the codebase of this.

**Comment by galderz** [comment]
@stuartwdouglas Updated

**Review by stuartwdouglas — APPROVED** [review]
Hmm, the validation must have gone missing somewhere. It's not a big deal, it will just end up generating an empty method so it is kinda pointless.

**Comment by geoand** [comment]
Should we merge this? Just asking because `waiting-for-ci` was added a few days ago and tests are passing

**Comment by dmlloyd** [comment]
@gsmet ?

**Comment by gsmet** [comment]
My review was just there to prevent merging it accidentally before addressing Stuart's concerns. Everyone seems to be happy now so let's merge.

**Pull request merged**